esinstall/src/types.ts

```diff
@@ -63,3 +64,31 @@ export type ExportMapEntry =
       default?: ExportMapEntry;
       require?: ExportMapEntry;
     };
+
+export type ExportMap = Record<string, ExportMapEntry>;
```

Comment:
Export maps may technically be a string as well:

```js
"exports": "./main.js"
```

Reply:
Oh actually you handle that in the function, and are testing for it. Ignore this comment. Typing this as a union type (`|`) would just cause more headaches. This WFM!

Reply:
I created `ExportField` type to handle the field itself. This mirrors the naming of BrowserField. ExportMap is the actual map (object) form.